contracts/EVMTokenBag.slb

```diff
@@ -0,0 +1,102 @@
+pragma solidity ^0.5.2;
+pragma experimental ABIEncoderV2;
+
+
+library EVMTokenBag {
+
+    struct Output {
+        address owner;
+        uint valueOrId;
+        bytes32 data;
+        address color;
+    }
+  
+    struct TokenBag {
+        Output[16] bag;
+    }
+    
+    function balanceOf(
+        TokenBag memory self,
+        address color,
+        address owner
+    ) internal pure returns (uint value) {
+        Output memory output;
+        for (uint i = 0; i < self.bag.length; i++) {
+            output = self.bag[i];
+            if (output.owner == owner && output.color == color) {
```

Comment:
even if we assume that the tokenbag is defragmented before, it can happen that it gets fragmented through `transfer()` during execution. either we have to build the defragmentation into `transfer()`, or assume fragmentation here, and have the loop run till the end and sum everything up?

Reply:
solved in call